### PR TITLE
Improved data collection for ASP.NET Core incoming requests

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**

### CODE REVIEW

File: src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs [hunk 1]
```diff
@@ -51,16 +55,26 @@ public override void OnStartActivity(Activity activity, object payload)
...
+
                 span.PutServerSpanKindAttribute();
-                span.PutHttpMethodAttribute(request.Method);
                 span.PutHttpHostAttribute(request.Host.Host, request.Host.Port ?? 80);
-                span.PutHttpPathAttribute(request.Path);
+                span.PutHttpMethodAttribute(request.Method);
+                span.PutHttpPathAttribute(path);
+
+                var userAgent = request.Headers["User-Agent"].FirstOrDefault();
+                span.PutHttpUserAgentAttribute(userAgent);
```
Comment (lmolkova) [inline src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs]: small optimization: don't put if it is null

File: src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs [hunk 2]
```diff
@@ -86,5 +100,45 @@ public override void OnStopActivity(Activity activity, object payload)
             span.PutHttpStatusCode(response.StatusCode, response.HttpContext.Features.Get<IHttpResponseFeature>().ReasonPhrase);
             span.End();
         }
+
+        private static string GetUri(HttpRequest request)
+        {
+            if (request == null)
+            {
+                return string.Empty;
```
Comment (lmolkova) [inline src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs]: is this ever a case on ASP.NET Core?

File: src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs [hunk 3]
```diff
@@ -86,5 +100,45 @@ public override void OnStopActivity(Activity activity, object payload)
...
+
+            if (request.Host.HasValue)
+            {
+                builder.Append(request.Host.Value);
+            }
+            else
+            {
+                // HTTP 1.0 request with NO host header would result in empty Host.
+                // Use placeholder to avoid incorrect URL like "http:///"
+                builder.Append(UnknownHostName);
```
Comment (lmolkova) [inline src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs]: is this ever a case?

File: src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs [hunk 4]
```diff
@@ -86,5 +100,45 @@ public override void OnStopActivity(Activity activity, object payload)
...
+            {
+                builder.Append(request.Path.Value);
+            }
+
+            if (request.QueryString.HasValue)
+            {
+                builder.Append(request.QueryString);
+            }
+
+            return builder.ToString();
```
Comment (lmolkova) [inline src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs]: why not request.ResuestUrl.ToString() (or whatever we have?)

Comment (SergeyKanzhelev) [reply src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs]: there is no request Url available. As far as I know

File: src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs (hunk 3 above)
Comment (SergeyKanzhelev) [reply src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs]: yes. See https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/278 fir discussion

File: src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs (hunk 2 above)
Comment (SergeyKanzhelev) [reply src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs]: nope. I'll remove

**Comment by SergeyKanzhelev** [comment]
@lmolkova I don't know better alternatives to the code I wrote. I'll remove the null check

**Pull request merged**